docs/create-packages/native-files-in-net-packages.md

```diff
@@ -0,0 +1,165 @@
+
+While NuGet has a [`contentFiles/` feature](../reference/msbuild-targets.md#including-content-in-a-package), it is not recommended to use this, as it doesn't support per-RID selection, and therefore doesn't participate in the .NET runtime's probing paths features.
+
+## Background
+
+First we'll consider a project referencing packages.
+When the app is published for a specific [Runtime Identifier (RID)](/dotnet/core/rid-catalog), then NuGet package assets for that RID will be copied into the output directory.
+When the app is published without a specific RID, all RID specific content will be published in subdirectories, and a `{project}.deps.json` file is created that the .NET runtime loads to determine which directory to use as a probing path.
+
+Also, be aware that .NET 8 introduces [breaking changes regarding how it determines compatible RIDs from the `.deps.json` file](/dotnet/core/compatibility/deployment/8.0/rid-asset-list).
+
+## Understanding NuGet package asset selection
+
+There are three types of assets in a NuGet file that are most relevant to this scenario, compile, runtime, and native assets.
+For a complete list of asset types, see the docs on [controlling dependency assets in `PackageReference` projects](../consume-packages/Package-References-in-Project-Files.md#controlling-dependency-assets).
+
+|Asset type|Short Description|
+|--|--|
+|[compile](#compile-assets)|Managed assemblies passed to the compiler. `refs/{tfm}/` if it exists, otherwise `lib/{tfm}/`.|
+|[runtime](#runtime-assets)|Managed assemblies copied to the output directory. `runtimes/{rid}/lib/{tfm}/` if it exists, otherwise `lib/{tfm}/`.|
+|[native](#native-assets)|Native libraries copied to the output directory. `runtimes/{rid}/native/`.|
+
+In all cases, NuGet chooses the "most compatible" directory for a given RID and Target Framework, and only the files in that directory are copied.
+Therefore, you cannot put common assets in an inherited RID (such as `any`), because if any other RID is a better match, the `any` RID contents will not be considered.
+
+To inspect the implementation of NuGet's asset selection, see [`NuGet.Client`'s `ManagedCodeConventions.cs` file](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/ContentModel/ManagedCodeConventions.cs).
+
+### Compile assets
+
+Compile assets are passed to the compiler, and therefore must only contain .NET assemblies.
+NuGet will select compile assets from `ref/{tfm}/`, and if that directory does not exist, it will fall back to `lib/{tfm}`, where `{tfm}` is the target framework that is the closest match the project referencing the package.
+However, in this scenario of creating a .NET package with native libraries, it is recommended to use `ref` because if `lib/` is used, NuGet will think the package is compatible with projects using `packages.config`, but those projects will fail at runtime because the native files will not be copied.
```

Comment:
nit:
```suggestion
However, in this scenario of creating a .NET package with native libraries, it is recommended to use `ref/` because if `lib/` is used, NuGet will think the package is compatible with projects using `packages.config`, but those projects will fail at runtime because the native files will not be copied.
```

for consistency.